### PR TITLE
Added iterator tests: for-each!,  count!, and find!

### PR DESCRIPTION
**Pull request by Izaakwltn**

### CODE REVIEW

File: tests/iterator-tests.lisp [hunk 1]
```diff
@@ -178,14 +178,35 @@
...
+    (is (== (iter:collect! (iter:into-iter (populate-vector)))
+	    (make-list 0 1 2 3 4)))))
+
+(define-test iter-count ()
+  (is (== (iter:count! (iter:into-iter (make-list 1 2 3 4 5)))
+	  5)))
+
+(define-test iter-find ()
+  (let ((even-in-iter (fn (iter)
+			(iter:find! (fn (x) (coalton-library/math/arith:zero? (mod x 2)))
```
Comment (eliaslfox) [inline tests/iterator-tests.lisp:198]: This should be written as `math:zero?`

**Review by eliaslfox — APPROVED** [review]
Thanks for adding tests

File: tests/iterator-tests.lisp (hunk 1 above)
Comment (eliaslfox) [reply tests/iterator-tests.lisp:198]: All of the `math/*` packages get pulled together [here](https://github.com/coalton-lang/coalton/blob/main/library/prelude.lisp#L5-L16).

**Review by stylewarning — COMMENTED** [review]
On mobile GitHub, indentation is still showing as being incorrect. @Izaakwltn

**Comment by eliaslfox** [comment]
I missed that. Maybe we should add a CI rule to ban tab characters from the repository.

**Pull request merged**